pkg/api/handlers/mcp.go

```diff
@@ -1428,6 +1655,149 @@ func (m *MCPHandler) ConfigureServer(req api.Context) error {
+	// Read configuration from request body
+	var configRequest struct {
+		ComponentConfigs map[string]struct {
+			Config   map[string]string `json:"config"`
+			URL      string            `json:"url"`
+			Disabled bool              `json:"disabled"`
+		} `json:"componentConfigs"`
+	}
+	if err := req.Read(&configRequest); err != nil {
+		return types.NewErrBadRequest("failed to read configuration: %v", err)
+	}
+
+	// Query component servers
+	var componentServers v1.MCPServerList
+	if err := req.List(&componentServers,
+		kclient.InNamespace(compositeServer.Namespace),
+		kclient.MatchingFields{"spec.compositeName": compositeServer.Name},
+	); err != nil {
+		return fmt.Errorf("failed to list component servers: %w", err)
+	}
+
+	// Load composite catalog entry once to map component manifests
+	var compositeEntry v1.MCPServerCatalogEntry
+	if err := req.Get(&compositeEntry, compositeServer.Spec.MCPServerCatalogEntryName); err != nil {
+		return fmt.Errorf("failed to get composite catalog entry %s: %w", compositeServer.Spec.MCPServerCatalogEntryName, err)
+	}
+
+	componentManifestByID := map[string]types.MCPServerCatalogEntryManifest{}
+	if compositeEntry.Spec.Manifest.CompositeConfig != nil {
+		for _, cs := range compositeEntry.Spec.Manifest.CompositeConfig.ComponentServers {
+			componentManifestByID[cs.CatalogEntryID] = cs.Manifest
+		}
+	}
+
+	// Configure each component with its env vars and persist enabled flags
+	// Build quick map for parent composite components by catalog entry ID
+	parentComps := map[string]int{}
+	if compositeServer.Spec.Manifest.CompositeConfig != nil {
+		for i, comp := range compositeServer.Spec.Manifest.CompositeConfig.ComponentServers {
+			parentComps[comp.CatalogEntryID] = i
+		}
+	}
+
+	for _, component := range componentServers.Items {
+		addExtractedEnvVars(&component)
+
+		// Calculate credential context for this component (after manifest update)
+		var (
+			credCtx = fmt.Sprintf("%s-%s", req.User.GetUID(), component.Name)
+			scope   = req.User.GetUID()
+		)
+
+		// Remove existing credential and shutdown server to allow update
+		if err := m.removeMCPServerAndCred(req.Context(), req.GPTClient, component, scope, []string{credCtx}); err != nil {
+			return err
+		}
+
+		componentConfig, hasConfig := configRequest.ComponentConfigs[component.Spec.MCPServerCatalogEntryName]
+		if !hasConfig {
+			continue
+		}
+
+		// Persist disabled state to parent composite manifest if present
+		if idx, ok := parentComps[component.Spec.MCPServerCatalogEntryName]; ok && compositeServer.Spec.Manifest.CompositeConfig != nil {
+			compositeServer.Spec.Manifest.CompositeConfig.ComponentServers[idx].Disabled = componentConfig.Disabled
+		}
+
+		if componentConfig.Disabled {
+			// On disable, clear remote URL (if any) and persist; credentials were already cleared above
+			if component.Spec.Manifest.RemoteConfig != nil {
+				component.Spec.Manifest.RemoteConfig.URL = ""
```

Comment:
If this URL is long and complicated, then the user would have to provide it again to enable right? I think it would be easier on our users if we had a different way of disabling this.

Reply:
do you have any suggestions there? I was thinking we could just leave the URL on the remote config when it's disabled.